onvif/src/discovery/mod.rs

```diff
@@ -186,20 +215,51 @@ where
                 .is_some()
         })
         .flat_map(|probe_match| {
-            probe_match
-                .x_addrs()
-                .into_iter()
-                .zip(iter::repeat(probe_match.name()))
+            let mut urls = probe_match.x_addrs();
+            let name = probe_match.name();
+
+            urls.retain(|url| match url.host() {
+                // Filter out link-local IP addresses.
+                // TODO: Make filtering link-local addresses configurable.
```

Comment:
I wonder if we should just expose this entire filtering logic so an application can provide it's own. This means that we wouldn't have to touch `onvif-rs` everytime we need changes of this logic in `lumeod` and other users of the library can then also decide to do it differently.

Reply:
Yeah, good point. Probably, it's better to move most of this out of `onvif-rs`.